vuu-ui/packages/vuu-ui-controls/src/date-picker/types.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  CalendarDate,
+  CalendarDateTime,
+  DateValue,
+  ZonedDateTime,
+} from "@internationalized/date";
+import { CalendarProps } from "../calendar/Calendar";
+import { RangeSelectionValueType } from "../calendar";
+
+export type PickerSelectionType = DateValue | RangeSelectionValueType;
+
+export interface BaseDatePickerProps<T = PickerSelectionType>
+  extends Pick<CalendarProps, "hideOutOfRangeDates" | "hideYearDropdown"> {
+  onSelectedDateChange: (selected: T) => void;
```

Comment:
probably good to keep APIs as similar as possible across form controls, so this could be just onSelectionChange to match list, dropdown,  combobox etc